SourceCode/HW2/HW2/HW2.cpp

```diff
@@ -0,0 +1,174 @@
+#include <fstream>
+#include <string>
+#include <array>
+#include <vector>
+#include <random>
+#include <numbers>
+#include <iostream>
+
+/// Output image resolution
+static constexpr uint32_t imageWidth = 1000;
+static constexpr uint32_t imageHeight = 1000;
+static constexpr uint32_t maxColorComponent = 255;
+
+class PPMWriter 
+{
+public:
+
+	PPMWriter(std::string filename, uint32_t imageWidth, uint32_t imageHeight, uint32_t maxColorComponent)
+		: ppmFileStream(filename + ".ppm", std::ios::out | std::ios::binary)
```

Comment:
It's a good idea to check if creation of the ofstream object succeeded. It could easily fail for example if you don't have write permission for the folder that the file is to be created in.

Reply:
Fixed in HW3 using exceptions.